samza-core/src/main/java/org/apache/samza/util/DiagnosticsUtil.java

```diff
@@ -70,7 +70,7 @@ public static void writeMetadataFile(String jobName, String jobId, String contai
       MetricsHeader metricsHeader =
           new MetricsHeader(jobName, jobId, "samza-container-" + containerId, execEnvContainerId.orElse(""),
               LocalContainerRunner.class.getName(), Util.getTaskClassVersion(config), Util.getSamzaVersion(),
-              Util.getLocalHost().getHostName(), System.currentTimeMillis(), System.currentTimeMillis());
+              Util.getLocalHost().getHostName(), System.currentTimeMillis(), System.currentTimeMillis(), config);
```

Comment:
Why do we want to add the config to the metric-header?
That way the entire config map (potentially 10s of MB of data) will be emitted in every message. 
This is NOT what we want to do right?
Instead we want to emit it in the MetricsMessage, but exactly once at container start (first message) not in every message.

Reply:
+1 I agree we should only pick the fields we are interested in. I suggest we create a versioned model (JSON serializable) that holds the fields we are interested in.

Reply:
Removed the config from the MetricsHeader.